scripts/doc-comments.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
```

Comment:
I always try to use `/bin/sh` after I caused a shell compatibility production incident 20 years ago.

However, we need bash features here and we know that CI provides bash so we are OK.

Reply:
If you insist on `sh` (it's fine either way):

```sh
#!/bin/sh

out=$(grep -nr "^--- @" lua)

if [ "$out" ]; then
	last_file=""
	printf %s "$out" | while read -r line; do
		file="$(echo "$line" | cut -d: -f1)"
		if [ "$file" != "$last_file" ]; then
			echo "$file:" >&2
			last_file="$file"
		fi
		echo "$line" | awk -F: '{ printf("  line %s: %s\n", $2, $3) }' >&2
	done
	exit 1
fi
```

Reply:
Works nicely, shellcheck is happy.

I will sleep better at night.